### PR TITLE
Jenkins

### PR DESCRIPTION
**Pull request by rowleya**
Makes integration testing work!

- [ ] Wait for merge of SpiNNakerManchester/sPyNNaker#576

Depends on merge of https://github.com/SpiNNakerManchester/PACMAN/pull/196

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        stage('Test') {
+            steps {
+                sh 'echo "<testsuite tests="0"></testsuite>" > results.xml'
+                sh 'py.test p8_integration_tests --forked --instafail --cov spynnaker8 --junitxml results.xml --timeout 1200'
+            }
+        }
+        stage('Coverage') {
+            steps {
+                sh 'COVERALLS_REPO_TOKEN=l0cQjQq6Sm5MGb67RiWkY2WE4r74YFAfk COVERALLS_PARALLEL=true coveralls'
```
Comment (dkfellows) [inline Jenkinsfile]: A world-readable magic token?

File: p8_integration_tests/test_sata_connectors/requires_virtual_board/test_sata_2_different_boards_no_board_address.py [hunk 2]
```diff
@@ -10,13 +10,14 @@ def do_run():
...
-                 label='External sata thing')
-
-    p.Population(2000, p.external_devices.ArbitraryFPGADevice,
-                 {'fpga_link_id': 11, 'fpga_id': 1, 'label': "bacon"},
-                 label='External sata thing')
+    p.Population(
+        None,
+        p.external_devices.ArbitraryFPGADevice(
+            2000, fpga_link_id=12, fpga_id=1, label="bacon"))
+
```
Comment (andrewgait) [inline p8_integration_tests/test_sata_connectors/requires_virtual_board/test_sata_2_different_boards_no_board_address.py]: @alan-stokes bacon has survived!  (So far...)

**Comment by Christian-B** [comment]
/sPyNNaker8/p8_integration_tests/test_various/test_stdp_example.py

post_spikes varies between 90 and 91 over multiple runs.
Reverted to range checking.

**Comment by Christian-B** [comment]
sPyNNaker8/p8_integration_tests/test_0_1_time_steps/test_va_bench_mark_tests/test_va_benchmark.py
Spike_count has also been seen as 2558

May need a range here too

**Comment by Christian-B** [comment]
spinnaker/sPyNNaker8/p8_integration_tests/test_csa_connectors/test_csa_connectors.py
fails with:
TypeError: 'float' object cannot be interpreted as an integer

**Comment by rowleya** [comment]
Is this a python 3 specific error?

**Comment by Christian-B** [comment]
Yes the csa problem is python3 specific because the latest csa release has a divide error.

Was fixed in Master on csa yesterday after we nudged him,

Requires one of csa master, a tweak to the release csa or python2.7 to work.

**Comment by Christian-B** [comment]
> Recent change to: p8_integration_tests/test_auto_pause_and_resume_tests/test_synfire_1_very_low_sdram.py
> 
> Seems that this only does one run on my system, and the same on Jenkins

This could be due to the number of chips available.  With more Chips the partitioner may spread it out more meaning there is less need to auto pause loop.

We should study
1. Why is the partitioner using more chips if it can?
2. If we agree the partitioner should use more chips how to make the test actaully work on both 48 and 4 chip boards

**Comment by Christian-B** [comment]
> Recent change to: p8_integration_tests/test_auto_pause_and_resume_tests/test_synfire_1_very_low_sdram.py
> 
> Seems that this only does one run on my system, and the same on Jenkins

Actually the issue is that the graph is split up into so many small vertexes that it does not fit on 4 chip board and does not need to auto pause.

Sdram_per_timestep prs fixes this so not wasting time here. Check commented out.

**Pull request merged**